tests/integration/fixtures/mini_sentry.py

```diff
@@ -225,10 +225,10 @@ def store_internal_error_event():
 
     @app.route("/api/42/envelope/", methods=["POST"])
     def store_event():
-        if flask_request.headers.get("Content-Encoding", "") == "gzip":
-            data = gzip.decompress(flask_request.data)
-        else:
-            abort(406, "Relay should always compress store requests")
+        assert (
+            flask_request.headers.get("Content-Encoding", "") == "gzip"
+        ), "Relay should always compress store requests"
+        data = gzip.decompress(flask_request.data)
```

Comment:
_nit_: For some reason I preferred the previous implementation.

Reply:
The previous implementation can only effectively be debugged with ngrok/wireshark to arrive at this error message.

Before:

> relay error: upstream returned NOT ACCEPTABLE

After:

> relay error: upstream returned INTERNAL SERVER ERROR
> teardown error: Relay should always compress store requests